src/pairing.cpp

```diff
@@ -148,13 +152,30 @@ Pair *check_good_pair(RoughReservoir* upper, RoughReservoir* lower,
   int head = upper->elevation - lower->elevation;
   double required_volume = find_required_volume(energy_capacity, head);
   if ((max(upper->volumes) < required_volume) ||
-      (max(lower->volumes) < required_volume))
+      (max(lower->volumes) < required_volume)) {
     return NULL;
+  }
+  ExistingPit single_pit;
+
+  if (search_config.search_type == SearchType::BULK_PIT) {
+    single_pit = pit_details[0];
```

Comment:
Do we need the `single_pit = pit_details[0]`? Seems like it could hide bugs

Reply:
Removed this line and added error handling if single_pit is not assigned a value